src/Stryker.Core/Stryker.Core/MutationTest/MutationTestProcess.cs

```diff
@@ -123,45 +121,54 @@ public void Mutate()
 
         private void CompileMutations()
         {
-            using (var ms = new MemoryStream())
+            using var ms = new MemoryStream();
+            using var msForSymbols = _options.DevMode ? new MemoryStream() : null;
+            // compile the mutated syntax trees
+            var compileResult = _compilingProcess.Compile(_input.ProjectInfo.ProjectContents.CompilationSyntaxTrees, ms, msForSymbols, _options.DevMode);
+
+            foreach (var testProject in _input.ProjectInfo.TestProjectAnalyzerResults)
             {
-                // compile the mutated syntax trees
-                var compileResult = _compilingProcess.Compile(_input.ProjectInfo.ProjectContents.CompilationSyntaxTrees, ms, _options.DevMode);
+                var injectionPath = testProject.TargetDirectory;
+                if (!_fileSystem.Directory.Exists(injectionPath))
+                {
+                    _fileSystem.Directory.CreateDirectory(injectionPath);
+                }
 
-                foreach (var testProject in _input.ProjectInfo.TestProjectAnalyzerResults)
+                // inject the mutated Assembly into the test project
+                using (var fs = _fileSystem.File.Create(Path.Combine(injectionPath, _input.ProjectInfo.ProjectUnderTestAnalyzerResult.TargetFileName)))
                 {
-                    var injectionPath = _input.ProjectInfo.GetInjectionPath(testProject);
-                    if (!_fileSystem.Directory.Exists(Path.GetDirectoryName(injectionPath)) &&
-                        !_fileSystem.File.Exists(injectionPath))
-                    {
-                        _fileSystem.Directory.CreateDirectory(Path.GetDirectoryName(injectionPath));
-                    }
+                    ms.Position = 0;
+                    ms.CopyTo(fs);
+                }
 
-                    // inject the mutated Assembly into the test project
-                    using (var fs = _fileSystem.File.Create(injectionPath))
+                if (msForSymbols != null)
+                {
+                    // inject the debug symbols into the test project
+                    using (var fs = _fileSystem.File.Create(Path.Combine(injectionPath,
```

Comment:
Same here

Reply:
Done, and renamed to avoid naming collision with previous